packages/core-transactions/src/handlers/transaction.ts

```diff
@@ -37,39 +37,37 @@ export abstract class TransactionHandler implements ITransactionHandler {
 
     public canBeApplied(
         transaction: Interfaces.ITransaction,
-        wallet: State.IWallet,
-        databaseWalletManager: State.IWalletManager,
-    ): boolean {
-        const { data }: Interfaces.ITransaction = transaction;
+        sender: State.IWallet,
+        databaseWalletManager?: State.IWalletManager,
+    ): void {
+        const data: Interfaces.ITransactionData = transaction.data;
+
+        if (Utils.isException(data)) {
+            return;
+        }
 
-        if (data.version > 1 && !data.nonce.isEqualTo(wallet.nonce.plus(1))) {
-            throw new UnexpectedNonceError(data.nonce, wallet.nonce.plus(1));
+        if (data.version > 1 && data.nonce.isLessThanOrEqualTo(sender.nonce)) {
+            throw new UnexpectedNonceError(data.nonce, sender.nonce, false);
         }
 
-        if (
-            wallet.balance
-                .minus(data.amount)
-                .minus(data.fee)
-                .isLessThan(0)
-        ) {
+        if (sender.balance.minus(data.amount).minus(data.fee).isNegative()) {
             throw new InsufficientBalanceError();
         }
 
-        if (data.senderPublicKey !== wallet.publicKey) {
+        if (data.senderPublicKey !== sender.publicKey) {
             throw new SenderWalletMismatchError();
         }
 
-        if (wallet.secondPublicKey) {
+        if (sender.secondPublicKey) {
             // Ensure the database wallet already has a 2nd signature, in case we checked a pool wallet.
-            const databaseWallet: State.IWallet = databaseWalletManager.findByPublicKey(
-                transaction.data.senderPublicKey,
-            );
-
-            if (!databaseWallet.secondPublicKey) {
-                throw new UnexpectedSecondSignatureError();
+            if (databaseWalletManager) {
```

Comment:
There should always be a wallet manager.

Reply:
Yes, indeed, good catch. Fixed in 5f3c02be1.